packages/vite/src/node/__tests__/plugins/workerImportMetaUrl.spec.ts

```diff
@@ -0,0 +1,161 @@
+import path from 'node:path'
+import { describe, expect, test } from 'vitest'
+import { parseAst } from 'rollup/parseAst'
+import { workerImportMetaUrlPlugin } from '../../plugins/workerImportMetaUrl'
+import { resolveConfig } from '../../config'
+import { PartialEnvironment } from '../../baseEnvironment'
+
+// @Note copied from packages/vite/src/shared/utils.ts
+const windowsSlashRE = /\\/g
+function slash(p: string): string {
+  return p.replace(windowsSlashRE, '/')
+}
+
+async function createWorkerImportMetaUrlPluginTransform() {
+  const root = slash(path.join(import.meta.dirname, 'fixtures/worker'))
+  const config = await resolveConfig({ configFile: false, root }, 'serve')
+  const instance = workerImportMetaUrlPlugin(config)
+  const environment = new PartialEnvironment('client', config)
+
+  return async (code: string) => {
+    // @ts-expect-error transform should exist
+    const result = await instance.transform.call(
+      { environment, parse: parseAst },
+      code,
+      path.join(root, 'foo.ts'),
+    )
+    return result?.code || result
+  }
+}
```

Comment:
```suggestion
async function createWorkerImportMetaUrlPluginTransform() {
  const config = await resolveConfig({ configFile: false }, 'serve')
  const instance = workerImportMetaUrlPlugin(config)
  const environment = new PartialEnvironment('client', config)

  return async (code: string) => {
    // @ts-expect-error transform should exist
    const result = await instance.transform.call(
      { environment, parse: parseAst },
      code,
      'foo.ts',
    )
    return result?.code || result
  }
}
```
It seems we can now simplify this and align with `assetImportMetaUrl.spec.ts`.

Reply:
Yes, we definitely can 👍